src/client/core/terminal/terminal.service.ts

```diff
@@ -0,0 +1,119 @@
+import { OSService } from "@batch-flask/ui/electron";
+import * as cp from "child_process";
+import { FileSystem } from "../fs";
+
+const supportedTerminals = {
+    powershell: {
+        process: "cmd.exe",
+        args: ["/c", "start", "powershell", "-NoExit", "-Command", "{command}"],
+    },
+    cmd: {
+        process: "cmd.exe",
+        args: ["/c", "start", "cmd", "/k", "{command}"],
+    },
+    linux: {
```

Comment:
here I was more thinking of listing all the linux terminals and depending on what the user choose get that.